test/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/RequestContentLengthUpdaterProcessorUnitTest.java

```diff
@@ -0,0 +1,247 @@
+    public void shouldCreateProcessorWithUndefinedMethod() {
+        // Given
+        String undefinedMethod = null;
+        // When
+        new RequestContentLengthUpdaterProcessor(undefinedMethod);
+        // Then = No Exception
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldFailToProcessAnUndefinedMessage() {
+        // Given
+        RequestContentLengthUpdaterProcessor processor = new RequestContentLengthUpdaterProcessor();
+        HttpMessage undefinedMessage = null;
+        // When
+        processor.processMessage(createUtils(), undefinedMessage);
+        // Then = NullPointerException
+    }
+
+    @Test
+    public void shouldNotRequireUtilsToProcessMessage() {
+        // Given
+        RequestContentLengthUpdaterProcessor processor = new RequestContentLengthUpdaterProcessor();
+        // When
+        processor.processMessage(null, new HttpMessage());
+        // Then = No Exception
+    }
+
+    @Test
+    public void shouldReturnSameMessageWhenProcessing() {
+        // Given
+        RequestContentLengthUpdaterProcessor processor = new RequestContentLengthUpdaterProcessor();
+        HttpMessage message = new HttpMessage();
+        // When
+        HttpMessage processedMessage = processor.processMessage(createUtils(), message);
+        // Then
+        assertThat(processedMessage, is(equalTo(message)));
+    }
+
+    @Test
+    public void shouldNotAddContentLengthIfEmptyBody() {
+        // Given
+        RequestContentLengthUpdaterProcessor processor = new RequestContentLengthUpdaterProcessor("POST");
+        HttpMessage messageEmptyBody = createHttpMessage("POST");
+        // When
+        processor.processMessage(createUtils(), messageEmptyBody);
+        // Then
+        assertThat(messageEmptyBody.getRequestHeader().getHeader(HttpRequestHeader.CONTENT_LENGTH), is(equalTo(null)));
+    }
+
+    @Test
+    public void shouldAddContentLengthIfNotEmptyBody() {
+        // Given
+        RequestContentLengthUpdaterProcessor processor = new RequestContentLengthUpdaterProcessor("POST");
+        HttpMessage messageEmptyBody = createHttpMessage("POST", null, "body");
+        // When
+        processor.processMessage(createUtils(), messageEmptyBody);
+        // Then
+        assertThat(messageEmptyBody.getRequestHeader().getHeader(HttpRequestHeader.CONTENT_LENGTH), is(equalTo("4")));
+    }
+
+    @Test
+    public void shouldUpdateExistingContentLengthIfEmptyBody() {
+        // Given
+        RequestContentLengthUpdaterProcessor processor = new RequestContentLengthUpdaterProcessor("POST");
+        HttpMessage messageEmptyBody = createHttpMessage("POST", 15);
+        // When
+        processor.processMessage(createUtils(), messageEmptyBody);
+        // Then
+        assertThat(messageEmptyBody.getRequestHeader().getHeader(HttpRequestHeader.CONTENT_LENGTH), is(equalTo("0")));
+    }
+
+    @Test
+    public void shouldUpdateContentLengthForAnyMethodWhenNoMethodIsSpecified() {
```

Comment:
All 3 specify a method? is `createHttpMessage("", 0, body);` valid/possible? Or is the function name misleading?